include/ck/utility/amd_wave_read_first_lane.hpp

```diff
@@ -95,11 +95,33 @@ using get_carrier_t = typename get_carrier<SizeInBytes>::type;
 
 } // namespace detail
 
+__device__ inline uint32_t amd_wave_read_first_lane(uint32_t value)
+{
+    return __builtin_amdgcn_readfirstlane(value);
+}
+
 __device__ inline int32_t amd_wave_read_first_lane(int32_t value)
 {
     return __builtin_amdgcn_readfirstlane(value);
 }
 
+__device__ inline int64_t amd_wave_read_first_lane(int64_t value)
+{
+    constexpr unsigned object_size        = sizeof(int64_t);
+    constexpr unsigned second_part_offset = object_size / 2;
+    auto* const from_obj                  = reinterpret_cast<const std::byte*>(&value);
+    alignas(int64_t) std::byte to_obj[object_size];
+
+    using Sgpr = uint32_t;
+
+    *reinterpret_cast<Sgpr*>(to_obj) =
```

Comment:
Any increasing scratch memory? For I know, using `reinterpret_cast()` is more likely introduce scrach memory usages.

Reply:
Or you can use `bit_cast`, which uses `memcpy `underneath.

Reply:
I investigated it and it is very unstable. Some cases have higher scratch for memcpy, some for reinterpret_cast (for most instances it is zero in both cases). The differences in scratch size are very small (from 4 to 20). I will try to investigate it more in my free time.